### PR TITLE
Rebase vshn v8.3.0

### PR DESCRIPTION
**Pull request by vanElden**
* Cherry-picked changes in voxpupuli up to v8.3.0
* Added focal support

### CODE REVIEW

**Review by andrekeller — COMMENTED** [review]
Most of the stuff seems to be "organizational" changes. If you test this on the an installation on each platform we use, I think it should be fine to merge.

This however adds back the totally broken deploy feature that stores the private keys in facts. I do not think we should merge this to vshn_master.

I'd remove that functionality in a separate commit.

**Comment by andrekeller** [comment]
Hmm this targets the master branch.

Shouldn't the master branch just follow master of the upstream module and changes would go into vshn_master?

**Comment by vanElden** [comment]
> Shouldn't the master branch just follow master of the upstream module and changes would go into vshn_master?

I had a closer look. We've been using the master branch for years now; the vshn_master branch is stale since 2017. This can of course be rebased, but would take quite some time. I'm not sure whether it's worth the effort.

**Comment by andrekeller** [comment]
The problem with using the master branch is that it is hard to follow upstream changes. Do we know what currently differs from our master and upstream master? It would also be an idea to just drop vshn_master, create a new vshn_master branch from our rebased master and then reset our master back to the upstream master.

**Comment by vanElden** [comment]
> Most of the stuff seems to be "organizational" changes. If you test this on the an installation on each platform we use, I think it should be fine to merge.

Tested and working fine afaics on Ubuntu Focal, Bionic, Xenial, CentOS 7 and CentOS 8, although the latter needs quite a few updates of other modules to get up to speed.

> This however adds back the totally broken deploy feature that stores the private keys in facts. I do not think we should merge this to vshn_master.

Good catch, this was re-introduced by the autolinter patch. Removed again.

**Comment by vanElden** [comment]
> It would also be an idea to just drop vshn_master, create a new vshn_master branch from our rebased master and then reset our master back to the upstream master.

That would likely be the way to go, as there are lots and lots of changes in the existing master. I guess we would have to:
. remove the existing `vshn_master` branch, add the new rebased one there
. tag accordingly
. move over environments (carefully. need to check what possibly breaks; probably do most of it in maintenance windows)
. once everything is moved to the new `vshn_master` branch / 8.3.0 tag drop the existing master and replace it with upstream

**Comment by andrekeller** [comment]
Sounds like a valid approach. Just make sure that the vshn_master branch is not used in any environment directly before you recreate it

**Comment by vanElden** [comment]
Changes done in `vshn_master` branch instead as discussed
